scripts/cronJobs.js

```diff
@@ -1,15 +1,117 @@
 const cron = require('node-cron');
-const userRecs = require('./simulateUserRecs.js');
-const genreRecs = require('./calculateGenreRecs');
+const request = require('request');
+const genreRecs = require('./calculateGenreRecs.js');
 
-const rand = () => Math.floor(Math.random() * 150) + 50;
+const genreArr = [
+  'action',
+  'animation',
+  'comedy',
+  'documentary',
+  'drama',
+  'family',
+  'fantasy',
+  'international',
+  'horror',
+  'musical',
+  'mystery',
+  'romance',
+  'sci_fi',
+  'thriller',
+  'western',
+];
 
-// simultate GET requests for individual user recommendations
-for (let i = 0; i < 10; i += 1) {
-  cron.schedule('* * * * *', () => {
-    setInterval(userRecs, rand());
+const processSessionData = () => {
+  const events = [];
+  for (let i = 0; i < 5; i += 1) {
+    events.push({
+      movie: {
+        id: Math.floor(Math.random() * 300000),
+      },
+      progress: Math.floor(Math.random() * 100) / 100,
+      timestamp: new Date(),
+    });
+  }
+  const sessionData = {
+    userId: Math.floor(Math.random() * 1000000),
+    events,
+  };
+  const options = {
+    url: 'http://localhost:3000/tetraflix/sessionData',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: sessionData,
+    json: true,
+  };
+  request(options, (error, response, body) => {
+    if (error) {
+      throw error;
+    }
   });
-}
+};
+
+const updateUserRecs = () => {
+  const recs = [];
+  for (let i = 0; i < 20; i += 1) {
+    recs.push(Math.floor(Math.random() * 300000));
+  }
+  const recsData = {
+    userId: Math.floor(Math.random() * 1000000),
+    rec: recs,
+  };
+  const options = {
+    url: 'http://localhost:3000/tetraflix/userRecs',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: recsData,
+    json: true,
+  };
+  request(options, (error, response, body) => {
+    if (error) {
+      throw error;
+    }
+  });
+};
+
+// CRON JOBS
 
 // update genre recommendations every hour
 cron.schedule('* 59 * * * *', genreRecs);
```

Comment:
I'm not 100% sure on the syntax, but you might be able to achieve the same timed intervals with '* * *\1 * *'. It's a minimal change, but if you want to make it, you may need to double check whether this is valid!

Reply:
I think that is a valid way of doing it.  There are definitely multiple ways of setting the timing.